actors/builtin/miner/miner_actor.go

```diff
@@ -298,71 +296,22 @@ func (a Actor) SubmitWindowedPoSt(rt Runtime, params *SubmitWindowedPoStParams)
 				params.Deadline, currEpoch, currDeadline.Index)
 		}
 
+		sectors, err := LoadSectors(store, st.Sectors)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load sectors")
+
 		deadline, err := deadlines.LoadDeadline(store, params.Deadline)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load deadline %d", params.Deadline)
 
-		partitions, err := deadline.PartitionsArray(store)
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load deadline %d partitions", params.Deadline)
-
 		faultExpiration := currDeadline.Last() + FaultMaxAge
+		postResult, err = deadline.ProcessWindowedPoSt(store, sectors, info.SectorSize, st.QuantEndOfDeadline(), faultExpiration, params.Partitions)
```

Comment:
The logic of handling all deadline state changes here is sound, but I find it a little uncomfortable declaring the post successful and sectors proven before validating the proof (knowing changes will be discarded if it fails). This should be called out here in a comment in `SubmitPoSt` since the logic is now buried.